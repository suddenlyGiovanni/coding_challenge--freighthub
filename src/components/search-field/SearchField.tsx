import * as React from 'react'
import TextField from '@material-ui/core/TextField'
import InputAdornment from '@material-ui/core/InputAdornment'
import IconButton from '@material-ui/core/IconButton'
import Search from '@material-ui/icons/Search'

interface Props {
  onSearchChange: (value: string) => void
  initialValue?: string
}

export const SearchField: React.FC<Props> = ({
  onSearchChange,
  initialValue = '',
}) => {
  const [value, setValue] = React.useState('')

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const _value = event.target.value
    setValue(_value)
    onSearchChange(_value)
  }

  return (
    <TextField
      id="search"
      label="Search"
      placeholder="Search"
      value={value}
      margin="none"
      onChange={handleChange}
      InputProps={{
        endAdornment: (
          <InputAdornment position="end">
            <IconButton edge="end" aria-label="search" disableTouchRipple>
              <Search />
            </IconButton>
          </InputAdornment>
        ),
      }}
    />
  )
}
