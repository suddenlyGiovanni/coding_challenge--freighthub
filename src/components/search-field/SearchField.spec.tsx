import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import { SearchField, Props } from './SearchField'

describe('SearchField', () => {
  const props: Props = {
    onSearchChange: value => {},
    initialValue: '',
  }

  test('snapshot', () => {
    // Arrange
    const { container } = render(<SearchField {...props} />)
    // Assert
    expect(container).toMatchSnapshot()
  })

  it('calls the `onSearchChange` when the user types', () => {
    // Arrange
    const search1 = 'U1000'
    const search2 = 'U1001'
    const handleOnSearchChange = jest.fn()
    const { getByLabelText } = render(
      <SearchField {...props} onSearchChange={handleOnSearchChange} />
    )
    const inputElement = getByLabelText('Search')

    // Act
    fireEvent.change(inputElement, { target: { value: search1 } })
    fireEvent.change(inputElement, { target: { value: search2 } })

    // Assert
    expect(handleOnSearchChange).toHaveBeenCalledTimes(2)
    expect(handleOnSearchChange).toHaveBeenCalledWith(search1)
    expect(handleOnSearchChange).toHaveBeenCalledWith(search2)
  })
})
