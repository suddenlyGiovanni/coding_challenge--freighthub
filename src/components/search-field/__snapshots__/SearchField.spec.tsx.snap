// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
      for="search"
    >
      Search
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
        id="search"
        placeholder="Search"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <button
          aria-label="search"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
